crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -107,9 +107,14 @@ impl SystemStage {
     }
 
     pub fn run_once(&mut self, world: &mut World, resources: &mut Resources) {
+        let mut system_refs = Vec::with_capacity(self.systems.len());
+        for system in self.systems.iter_mut() {
+            system_refs.push(system.as_mut());
+        }
```

Comment:
This would allocate a new vec every time the stage runs, right? Maybe pass an iterator instead?